tm1638/tm1638.go

```diff
@@ -0,0 +1,137 @@
+/*
+TM1638 is an LED Controller
+*/
+package tm1638
+
+import (
+	"machine"
+	"time"
+)
+
+// Device wraps the pins of the TM1638.
+type Device struct {
+	/* STB - When this pin is "LO" chip accepting transmission */
+	strobe machine.Pin
+	/* CLK - DIO pin reads data at the rising edge and outputs at the falling edge */
+	clock machine.Pin
+	/* DIO - This pin outputs/inputs serial data */
+	data machine.Pin
+}
+
+const (
+	addressAutoIncrement = 0x40
+	fixedAddress         = 0x44
+	baseAddress          = 0xC0
+	maxAddress           = 0x0F
+	readKeys             = 0x42
+	brightness           = 0x88
+	maxBrightness        = 0x07
+	zeroBrightness       = 0x80
+)
+
+// Create new TM1638 device
+func New(strobe machine.Pin, clock machine.Pin, data machine.Pin) Device {
+	strobe.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	clock.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	return Device{strobe: strobe, clock: clock, data: data}
+}
+
+// Configure TM1638
+func (d *Device) Configure() {
```

Comment:
Can you add a `Config` parameter?

```go
type Config struct {
    Brightness uint8
}
```

Now it's only for the Brightness value, but it may be useful for other configuration later.

Reply:
I will add this parameter.

Is it good idea also  to add `*Config` field in `Device`?
```go
type Device struct {
	strobe machine.Pin
	clock machine.Pin
	data machine.Pin
	config *Config // Is it good idea?
}
```

In this way I can add state and encapsulate functions like
```go
// Get actual brightness
func (d *Device) Brigtness() uint8 {
}
// Increase brightness level by 1
func (d *Device) IncreaseBrigtness() {
}
// Decrease brightness level by 1
func (d *Device) DecreaseBrigtness() {
}
```